api/src/controllers/people.js

```diff
@@ -50,7 +68,13 @@ const validatePerson = obj => {
  * validate_doc_update. https://github.com/medic/medic/issues/2203
  */
 const createPerson = data => {
-  data.type = 'person';
+  if (!data.type) {
+    const defaultType = getDefaultPersonType();
+    if (defaultType) {
+      data.type = defaultType;
+    }
+    // else validation will fail below
+  }
```

Comment:
The default person type is "person" which is one of the old hardcoded types. The only options for that function to return are `undefined` or `"person"` so I think it's ok, but perhaps not very readable. Does that make sense? I could change it to `hasHardcodedPersonType` which returns a boolean or something?

Reply:
What I meant is that maybe we should invalidate records that use a "flexible" id as `type`, instead of doing `contact_type` + `type`. 
I was also confused as to why we create new records with hardcoded types by default, but that's clear now.

Reply:
I could add some validation so if `type` is specified then it must be `contact` and there must be a `contact_type` with the ID of a person type. If any of that is not right we can just fail. Does that make sense?

Reply:
Yep! Cool!

Reply:
Done.